src/main/java/com/cloudogu/scm/ci/cistatus/service/CIStatusService.java

```diff
@@ -41,20 +41,23 @@ public CIStatusService(DataStoreFactory dataStoreFactory) {
     this.dataStoreFactory = dataStoreFactory;
   }
 
-  public void put(Repository repository, String changesetId, CIStatus ciStatus) {
+  public void put(String storeName, Repository repository, String changesetId, CIStatus ciStatus) {
     PermissionCheck.checkWrite(repository);
-    CIStatusCollection ciStatusCollection = get(repository, changesetId);
+    CIStatusCollection ciStatusCollection = get(storeName, repository, changesetId);
     ciStatusCollection.put(ciStatus);
-    getStore(repository).put(changesetId, ciStatusCollection);
+    getStore(storeName, repository).put(changesetId, ciStatusCollection);
   }
 
-  public CIStatusCollection get(Repository repository, String changesetId) {
+  public CIStatusCollection get(String storeName, Repository repository, String changesetId) {
```

Comment:
For the time being I would use an enum for the store (with a text field for the concrete name), so that one cannot mistakenly use other strings here.